sdk/src/test/java/io/dapr/client/DaprClientGrpcTest.java

```diff
@@ -167,44 +162,29 @@ public void publishEventSerializeException() throws IOException {
 
   @Test
   public void publishEventTest() {
-    SettableFuture<Empty> settableFuture = SettableFuture.create();
-    MockCallback<Empty> callback = new MockCallback<>(Empty.newBuilder().build());
-    addCallback(settableFuture, callback, directExecutor());
-    when(client.publishEvent(any(DaprProtos.PublishEventRequest.class)))
-        .thenReturn(settableFuture);
+    doAnswer((Answer<Void>) invocation -> {
+      StreamObserver<Empty> observer = (StreamObserver<Empty>) invocation.getArguments()[1];
+      observer.onNext(Empty.getDefaultInstance());
+      observer.onCompleted();
+      return null;
+    }).when(client).publishEvent(any(DaprProtos.PublishEventRequest.class), any());
+
     Mono<Void> result = adapter.publishEvent("pubsubname","topic", "object");
-    settableFuture.set(Empty.newBuilder().build());
     result.block();
-    assertTrue(callback.wasCalled);
-  }
-
-  @Test
-  public void publishEventNoHotMono() {
```

Comment:
Please, bring back all the hot mono tests. These are critical.

Reply:
I do not see the point of unit testing "hot" mono for every scenery. Once get familiar with reactive programming, one would know that nothing happens until a stream is subscribed.

Reply:
The problem was that the code was actually invoking a service call and just wrapping the result with Mono.just(). These tests are here to make sure this issue does not come back. I prefer to have the test over trusting that everyone touching this code going forward will avoid this bug.

Reply:
Ok. Added "NoHotMono" unit tests